pkgs/tools/archivers/7zz/default.nix

```diff
@@ -1,47 +1,45 @@
-{ lib, stdenv, fetchurl, autoPatchelfHook }:
+{ stdenv, lib, fetchurl, gnugrep, p7zip }:
 
-let platform =       if stdenv.isi686    then "x86"
-                else if stdenv.isx86_64  then "x64"
-                else if stdenv.isAarch32 then "arm"
-                else if stdenv.isAarch64 then "arm64"
-                else throw "Unsupported architecture";
+# https://sourceforge.net/p/sevenzip/discussion/45797/thread/7fe6c21efa/
+stdenv.mkDerivation rec {
+  pname = "7zz";
+  version = "21.02";
 
-    url = "https://7-zip.org/a/7z2101-linux-${platform}.tar.xz";
+  src = fetchurl {
+    url = "https://7-zip.org/a/7z${lib.replaceStrings ["." ] [""] version}-src.7z";
+    sha256 = "sha256-7pdV/qoDTnB1cSvGsfrSVEC9Co8h8Qahbje8S0CfESI=";
+  };
 
-    hashes = {
-      x86 = "0k6vg85ld8i2pcv5sv3xbvf3swqh9qj8hf2jcpadssys3yyidqyj";
-      x64 = "1yfanx98fizj8d2s87yxgsy30zydx7h5w9wf4wy3blgsp0vkbjb3";
-      arm = "04iah9vijm86r8rbkhxig86fx3lpag4xi7i3vq7gfrlwkymclhm1";
-      arm64 = "0a26ginpb22aydcyvffxpbi7lxh4sgs9gb6cj96qqx7cnf7bk2ri";
-    };
-    sha256 = hashes."${platform}";
+  sourceRoot = "CPP/7zip/Bundles/Alone2";
 
-in stdenv.mkDerivation {
-  pname = "7zz";
-  version = "21.01";
+  # we need https://github.com/nidud/asmc/tree/master/source/asmc/linux in order
+  # to build with the optimized assembler but that doesn't support building with
+  # GCC
+  # "../../cmpl_gcc_x64.mak";
+  makefile = "../../cmpl_gcc.mak";
 
-  src = fetchurl { inherit url sha256; };
-  sourceRoot = ".";
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=maybe-uninitialized" ];
 
-  nativeBuildInputs = [ autoPatchelfHook ];
-  buildInputs = [ stdenv.cc.cc.lib ];
+  nativeBuildInputs = [ gnugrep p7zip ];
 
-  dontBuild = true;
+  enableParallelBuilding = true;
 
   installPhase = ''
-    runHook preInstall
-    install -D -t $out/bin 7zz
-    runHook postInstall
+    install -Dm555 -t $out/bin b/g/7zz
+    install -Dm444 -t $out/share/doc/${pname} ../../../../DOC/*.txt
   '';
 
-  meta = with lib; {
-    description = "Command line archiver utility";
-    homepage = "https://www.7-zip.org";
+  doInstallCheck = true;
 
-    # source not released yet. will be under LGPL 2.1+ with RAR exception
-    license = licenses.unfree;
+  installCheckPhase = ''
+    $out/bin/7zz --help | grep ${version}
+  '';
 
-    platforms = [ "i686-linux" "x86_64-linux" "aarch64-linux" "armv7l-linux" ];
-    maintainers = with maintainers; [ anna328p ];
+  meta = with lib; {
+    description = "Command line archiver utility";
+    homepage = "https://7zip.org";
+    license = licenses.lgpl21Only;
```

Comment:
LICENSE.txt

<details>

```
  7-Zip source code
  ~~~~~~~~~~~~~~~~~
  License for use and distribution
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  7-Zip Copyright (C) 1999-2020 Igor Pavlov.

  The licenses for files are:

    1) CPP/7zip/Compress/Rar* files: the "GNU LGPL" with "unRAR license restriction"
    2) CPP/7zip/Compress/LzfseDecoder.cpp: the "BSD 3-clause License"
    3) Some files are "public domain" files, if "public domain" status is stated in source file.
    4) the "GNU LGPL" for all other files. If there is no license information in 
       some source file, that file is under the "GNU LGPL".

  The "GNU LGPL" with "unRAR license restriction" means that you must follow both 
  "GNU LGPL" rules and "unRAR license restriction" rules.




  GNU LGPL information
  --------------------

    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Lesser General Public
    License as published by the Free Software Foundation; either
    version 2.1 of the License, or (at your option) any later version.

    This library is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.

    You should have received a copy of the GNU Lesser General Public
    License along with this library; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA




  BSD 3-clause License
  --------------------

    The "BSD 3-clause License" is used for the code in LzfseDecoder.cpp that implements LZFSE data decompression.
    That code was derived from the code in the "LZFSE compression library" developed by Apple Inc,
    that also uses the "BSD 3-clause License":

    ----
    Copyright (c) 2015-2016, Apple Inc. All rights reserved.

    Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

    1.  Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

    2.  Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer
        in the documentation and/or other materials provided with the distribution.

    3.  Neither the name of the copyright holder(s) nor the names of any contributors may be used to endorse or promote products derived
        from this software without specific prior written permission.

    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
    COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
    HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
    ----




  unRAR license restriction
  -------------------------

    The decompression engine for RAR archives was developed using source 
    code of unRAR program.
    All copyrights to original unRAR code are owned by Alexander Roshal.

    The license for original unRAR code has the following restriction:

    The unRAR sources cannot be used to re-create the RAR compression algorithm, 
    which is proprietary. Distribution of modified unRAR sources in separate form 
    or as a part of other software is permitted, provided that it is clearly
    stated in the documentation and source comments that the code may
    not be used to develop a RAR (WinRAR) compatible archiver.


  --
  Igor Pavlov
```
</details>

> This library is free software; you can redistribute it and/or
> modify it under the terms of the GNU Lesser General Public
> License as published by the Free Software Foundation; either
> version 2.1 of the License, or (at your option) any later version.

Seems to me `licenses.lgpl21` instead.

Reply:
Sorry, `licenses.lgpl21Plus`.